airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -227,44 +399,162 @@ protected void putTimestamp(final ObjectNode node, final String columnName, fina
+        case "_name" -> PostgresType.NAME_ARRAY;
+        case "_varchar" -> PostgresType.VARCHAR_ARRAY;
+        case "_char" -> PostgresType.CHAR_ARRAY;
+        case "_bpchar" -> PostgresType.BPCHAR_ARRAY;
+        case "_text" -> PostgresType.TEXT_ARRAY;
+        case "_int4" -> PostgresType.INT4_ARRAY;
+        case "_int2" -> PostgresType.INT2_ARRAY;
+        case "_int8" -> PostgresType.INT8_ARRAY;
+        case "_money" -> PostgresType.MONEY_ARRAY;
+        case "_oid" -> PostgresType.OID_ARRAY;
+        case "_numeric" -> PostgresType.NUMERIC_ARRAY;
+        case "_float4" -> PostgresType.FLOAT4_ARRAY;
+        case "_float8" -> PostgresType.FLOAT8_ARRAY;
+        case "_timestamptz" -> PostgresType.TIMESTAMPTZ_ARRAY;
+        case "_timestamp" -> PostgresType.TIMESTAMP_ARRAY;
+        case "_timetz" -> PostgresType.TIMETZ_ARRAY;
+        case "_time" -> PostgresType.TIME_ARRAY;
+        case "_date" -> PostgresType.DATE_ARRAY;
+        case "_bytea" -> PostgresType.BYTEA_ARRAY;
+        case "bool", "boolean" -> PostgresType.BOOLEAN;
         // BYTEA is variable length binary string with hex output format by default (e.g. "\x6b707a").
         // It should not be converted to base64 binary string. So it is represented as JDBC VARCHAR.
         // https://www.postgresql.org/docs/14/datatype-binary.html
-        case "bytea" -> JDBCType.VARCHAR;
-        case TIMESTAMPTZ -> JDBCType.TIMESTAMP_WITH_TIMEZONE;
-        case TIMETZ -> JDBCType.TIME_WITH_TIMEZONE;
-        default -> JDBCType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt());
+        case "bytea" -> PostgresType.VARCHAR;
+        case TIMESTAMPTZ -> PostgresType.TIMESTAMP_WITH_TIMEZONE;
+        case TIMETZ -> PostgresType.TIME_WITH_TIMEZONE;
+        default -> PostgresType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt());
       };
     } catch (final IllegalArgumentException ex) {
       LOGGER.warn(String.format("Could not convert column: %s from table: %s.%s with type: %s. Casting to VARCHAR.",
           field.get(INTERNAL_COLUMN_NAME),
           field.get(INTERNAL_SCHEMA_NAME),
           field.get(INTERNAL_TABLE_NAME),
           field.get(INTERNAL_COLUMN_TYPE)));
-      return JDBCType.VARCHAR;
+      return PostgresType.VARCHAR;
     }
   }
 
   @Override
-  public JsonSchemaType getJsonType(final JDBCType jdbcType) {
+  public JsonSchemaType getJsonType(final PostgresType jdbcType) {
     return switch (jdbcType) {
       case BOOLEAN -> JsonSchemaType.BOOLEAN;
       case TINYINT, SMALLINT, INTEGER, BIGINT -> JsonSchemaType.INTEGER;
       case FLOAT, DOUBLE, REAL, NUMERIC, DECIMAL -> JsonSchemaType.NUMBER;
       case BLOB, BINARY, VARBINARY, LONGVARBINARY -> JsonSchemaType.STRING_BASE_64;
       case ARRAY -> JsonSchemaType.ARRAY;
+      case BIT_ARRAY -> JsonSchemaType.builder(JsonSchemaPrimitive.ARRAY)
+              .withItems(JsonSchemaType.builder(JsonSchemaPrimitive.BOOLEAN)
+                      .build())
+              .withAirbyteType("bit_array")
```

Comment:
We shouldn't be defining new `airbyte_type` values like this. Each airbyte_type should represent a primitive type; complex types (such as arrays) should be defined using the jsonschema equivalent, e.g.
```json
{
  "type": "array",
  "items": {
    "type": "string",
    "airbyte_type": "timestamp_with_timezone"
  }
}
```

Note that the top-level JsonSchema (type:array) does _not_ declare airbyte_type; only the nested type does.

Reply:
> We shouldn't be defining new `airbyte_type` values like this. Each airbyte_type should represent a primitive type; complex types (such as arrays) should be defined using the jsonschema equivalent, e.g.
> 
> ```json
> {
>   "type": "array",
>   "items": {
>     "type": "string",
>     "airbyte_type": "timestamp_with_timezone"
>   }
> }
> ```
> 
> Note that the top-level JsonSchema (type:array) does _not_ declare airbyte_type; only the nested type does.

@edgao i have updated json schema mapping

Reply:
sorry, to clarify - this is what it should look like:
```java
      case BIT_ARRAY -> JsonSchemaType.builder(JsonSchemaPrimitive.ARRAY)
              .withItems(JsonSchemaType.builder(JsonSchemaPrimitive.BOOLEAN).build())
              .build();
      // (skipping some entries...)
      case INT8_ARRAY -> JsonSchemaType.builder(JsonSchemaPrimitive.ARRAY)
              .withItems(JsonSchemaType.builder(JsonSchemaPrimitive.NUMBER)
                      .withAirbyteType("integer")
                      .build())
              .build();
```

Which is equivalent to these JSON schemas:
```json
{
  "type": "array",
  "items": {
    "type": "boolean"
  }
}
```
```json
{
  "type": "array",
  "items": {
    "type": "number",
    "airbyte_type": "integer"
  }
}
```

The only valid airbyte_type values are the ones already used in JsonSchemaType. Defining new values for them is a protocol change, and would require additional work beyond just this PR (e.g. updating normalization to handle them).

(fyi: airbyte_type is going away, and will be replaced by https://github.com/airbytehq/airbyte/pull/17486. But the basic statement remains the same: we shouldn't define new types without significant discussion)

Reply:
@edgao updated json schema mapping